controllers/release/release_adapter.go

```diff
@@ -0,0 +1,301 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package release
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	libhandler "github.com/operator-framework/operator-lib/handler"
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/controllers/results"
+	"github.com/redhat-appstudio/release-service/tekton"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"time"
+)
+
+// Adapter holds the objects needed to reconcile a Release.
+type Adapter struct {
+	release *v1alpha1.Release
+	logger  logr.Logger
+	client  client.Client
+	context context.Context
+}
+
+// finalizerName is the finalizerName name to be added to the Releases
+const finalizerName string = "appstudio.redhat.com/release-finalizerName"
+
+// NewAdapter creates and returns an Adapter instance.
+func NewAdapter(release *v1alpha1.Release, logger logr.Logger, client client.Client, context context.Context) *Adapter {
+	return &Adapter{
+		release: release,
+		logger:  logger,
+		client:  client,
+		context: context,
+	}
+}
+
+// EnsureFinalizersAreCalled is an operation that will ensure that finalizers are called whenever the Release being
+// processed is marked for deletion. Once finalizers get called, the finalizerName will be removed and the Release will go
+// back to the queue, so it gets deleted.
```

Comment:
`gets deleted. If a finalizer fails its execution or a finalizer fails to be removed, the Release will be requeued with the error attached`

Reply:
Done.